pil_analyzer/src/pil_analyzer.rs

```diff
@@ -110,27 +110,14 @@ impl<T: FieldElement> PILAnalyzer<T> {
         }
     }
 
-    pub fn process(&mut self, files: Vec<(PathBuf, PILFile<T>)>) {
+    pub fn process(&mut self, files: Vec<PILFile<T>>) {
         self.current_namespace = Default::default();
-        for statement in files.iter().flat_map(|(_, f)| f.0.iter()) {
+        for statement in files.iter().flat_map(|f| f.0.iter()) {
             self.collect_names(statement);
         }
 
         self.current_namespace = Default::default();
```

Comment:
I think resetting the namespace at the beginning of each file instead of at the beginning of processing is the right thing to do. It is a fix and not a refactoring, but I think we should do it in this PR nevertheless.

Reply:
(same for the collect_names loop above)